frontend/src/pages/notebookController/screens/admin/NotebookActions.tsx

```diff
@@ -1,14 +1,22 @@
 import * as React from 'react';
 import { ActionsColumn, IAction } from '@patternfly/react-table';
+import StopServerModal from '~/pages/notebookController/screens/server/StopServerModal';
+import { useStopWorkbenchModal } from '~/concepts/notebooks/useStopWorkbenchModal';
 import { AdminViewUserData } from './types';
-import { NotebookAdminContext } from './NotebookAdminContext';
 
 type ServerStatusProps = {
   data: AdminViewUserData['actions'];
 };
 
 const NotebookActions: React.FC<ServerStatusProps> = ({ data }) => {
-  const { setServerStatuses } = React.useContext(NotebookAdminContext);
+  const notebooksToStop = React.useMemo(() => (data.notebook ? [data.notebook] : []), [data]);
+
+  const { showModal, isDeleting, onStop, onNotebooksStop } = useStopWorkbenchModal({
+    notebooksToStop,
+    refresh: () => {
+      data.forceRefresh();
```

Comment:
You are creating an unstable function as a dependency in your hook. Memoize it by using `React.useCallback`